components/mpas-ocean/src/shared/mpas_ocn_vmix.F

```diff
@@ -1263,26 +1276,39 @@ subroutine ocn_vmix_implicit(dt, meshPool, statePool, forcingPool, scratchPool,
 #ifdef MPAS_OPENACC
       !$acc enter data copyin(layerThickness, normalVelocity)
 #endif
-      if (config_use_implicit_bottom_drag_variable) then
-        call ocn_vel_vmix_tend_implicit_spatially_variable(bottomDrag, dt, kineticEnergyCell, &
+      select case (dragType)
+
+      case (dragTypeConst)
+
+        call ocn_vel_vmix_tend_implicit(dt, kineticEnergyCell, &
+          vertViscTopOfEdge, layerThickness, layerThickEdgeMean, &
+          landIceEdgeFraction, normalVelocity, err)
+
+      case (dragTypeRayleigh)
+
+        call ocn_vel_vmix_tend_implicit_rayleigh(dt, kineticEnergyCell, &
           vertViscTopOfEdge, layerThickness, layerThickEdgeMean, normalVelocity, err)
-      else if (config_use_implicit_bottom_drag_variable_mannings.or. &
-               config_use_implicit_bottom_roughness) then
-        ! update bottomDrag via Cd=g*n^2*h^(-1/3)
+
+      case (dragTypeLogLaw)
+
         call ocn_vel_vmix_tend_implicit_spatially_variable_mannings(forcingPool, bottomDrag, &
```

Comment:
Should we remove `mannings` from this subroutine to avoid confusion, since it supports both the mannings and log-law options?

Reply:
I'd like to leave it as-is because after this is merged I am planning to open (and rebase) the fix to log-law https://github.com/E3SM-Ocean-Discussion/E3SM/pull/40 which moves the log-law formulation to its own subroutine:

https://github.com/E3SM-Ocean-Discussion/E3SM/blob/16a438752051aada9933d0794360362da595b9f3/components/mpas-ocean/src/shared/mpas_ocn_vmix.F#L940

Reply:
That sounds good to me.